test/integration/StreamingHttpChannel2023.test.ts

```diff
@@ -0,0 +1,309 @@
+import { DataFactory, Parser, Store } from 'n3';
+import { BasicRepresentation } from '../../src/http/representation/BasicRepresentation';
+import type { App } from '../../src/init/App';
+import type { ResourceStore } from '../../src/storage/ResourceStore';
+import { joinUrl } from '../../src/util/PathUtil';
+import { AS, RDF } from '../../src/util/Vocabularies';
+import { getPort } from '../util/Util';
+import {
+  getDefaultVariables,
+  getPresetConfigPath,
+  getTestConfigPath,
+  getTestFolder,
+  instantiateFromConfig,
+  removeFolder,
+} from './Config';
+import namedNode = DataFactory.namedNode;
+
+const port = getPort('StreamingHTTPChannel2023');
+const baseUrl = `http://localhost:${port}/`;
+
+const rootFilePath = getTestFolder('StreamingHTTPChannel2023');
+const stores: [string, any][] = [
+  [ 'in-memory storage', {
+    configs: [ 'storage/backend/memory.json', 'util/resource-locker/memory.json' ],
+    teardown: jest.fn(),
+  }],
+  [ 'on-disk storage', {
+    configs: [ 'storage/backend/file.json', 'util/resource-locker/file.json' ],
+    teardown: async(): Promise<void> => removeFolder(rootFilePath),
+  }],
+];
+
+describe.each(stores)('A server supporting StreamingHTTPChannel2023 using %s', (name, { configs, teardown }): void => {
+  let app: App;
+  let store: ResourceStore;
+  const webId = 'http://example.com/card/#me';
+  const topic = joinUrl(baseUrl, '/foo');
+  const pathPrefix = '.notifications/StreamingHTTPChannel2023';
+  const receiveFrom = joinUrl(baseUrl, pathPrefix, '/foo');
+
+  beforeAll(async(): Promise<void> => {
+    const variables = {
+      ...getDefaultVariables(port, baseUrl),
+      'urn:solid-server:default:variable:rootFilePath': rootFilePath,
+    };
+
+    // Create and start the server
+    const instances = await instantiateFromConfig(
+      'urn:solid-server:test:Instances',
+      [
+        ...configs.map(getPresetConfigPath),
+        getTestConfigPath('streaming-http-notifications.json'),
+      ],
+      variables,
+    ) as Record<string, any>;
+    ({ app, store } = instances);
+
+    await app.start();
+  });
+
+  afterAll(async(): Promise<void> => {
+    await teardown();
+    await app.stop();
```

Comment:
It doesn't resolve on node 21
I will focus on the other cleanup in case you have an idea of the problem. Otherwise, I will return to it as the last step in wrapping up this PR.

Reply:
It's unfortunate that it only happens on Node 21. Might mean it is just the consequence of an issue in that version that could get fixed in the future. You might want to play around with disabling tests in the suite to see if it is one specific test that is the cause. I remember running into similar issues when implementation the other notification types because a connection wasn't closed causing the tests to not halt.

Reply:
https://github.com/CommunitySolidServer/CommunitySolidServer/blob/f59b2c29704f48c02bcae13787588322524b2511/src/init/ServerInitializer.ts#L46
Adding `this.server.closeAllConnections()` before this line works, so it looks like, in node >21, some requests in the test are still awaiting a response.

Reply:
I found a workaround 340c6d385. Setting a timeout in the test for the request body ensures that all connections get closed. I will still follow up on it in the `undici` repo to find out what has changed between node 20 and 21 when calling `response.body.close()`

Reply:
There was a bug in `undici,` which has already been fixed. Node 22.1 already works as expected, and eventually, the newer version of `undici` will make its way to node 21, and we will be able to remove those body timeouts.